typescript/scripts/eval/promptfoo/run_aiconfig.ts

```diff
@@ -0,0 +1,35 @@
+import { config as load_dotenv } from "dotenv";
+import { AIConfigRuntime } from "../../../lib/config";
+import fs from "fs";
+import path from "path";
+
+async function main() {
+  let settings_path = process.argv[2];
+  let question = process.argv[3];
+  let settings = _load_settings(settings_path);
+  let prompt_name = settings["prompt_name"];
+  let aiconfig_path = settings["aiconfig_path"];
+  load_dotenv();
+
+  let fullAIConfigPath = path.join(__dirname, aiconfig_path);
+  let runtime = AIConfigRuntime.load(fullAIConfigPath);
+  let params = {
+    the_query: question,
+  };
+  let result = await runtime.run(prompt_name, params);
+  let final_output: string;
+
+  let r0 = Array.isArray(result) ? result[0] : result;
+  final_output = runtime.getOutputText(prompt_name, r0);
+  console.log(final_output);
+}
+
+function _load_settings(settings_path: string) {
+  let rawdata = fs.readFileSync(settings_path, "utf-8");
+  let settings = JSON.parse(rawdata);
+  return settings;
+}
+
+if (process.env.npm_lifecycle_event != null) {
+  main();
+}
```

Comment:
What is this w/ process.env.npm_lifecycle_event and why is it needed? (I've never seen this used before btw, why can't I just call `main()`?).